numpyro/util.py

```diff
@@ -382,7 +382,7 @@ def soft_vmap(fn, xs, batch_ndims=1, chunk_size=None):
 
     # we'll do map(vmap(fn), xs) and make xs.shape = (num_chunks, chunk_size, ...)
     num_chunks = batch_size = int(np.prod(batch_shape))
-    prepend_shape = (-1,) if batch_size > 1 else ()
+    prepend_shape = (batch_size,) if batch_size > 1 else ()
```

Comment:
@fehiepsi i guess these changes are leading to some failing tests, for example:

test/infer/test_infer_util.py::test_log_likelihood[batch_shape0] FAILED
test/test_util.py::test_soft_vmap[None-batch_shape0] FAILED
test/test_util.py::test_soft_vmap[None-batch_shape1] FAILED

Reply:
fixed, `np.prod(...)` returns a float value?? I just convert it to `int`.